### PR TITLE
Impement new question modal design

### PR DESCRIPTION
**Pull request by musayann**
Integrated the new question design with the vue frontend. there is still room for improvement, i think we have to get the iteration completion rate in the vue modal from the api to update the progress bar

### CODE REVIEW

File: app/javascript/views/Question.vue [hunk 1]
```diff
@@ -1,12 +1,33 @@
...
+            </div>
+        </div>
+        <div class="w-100 border-0 border-b-1 border-solid"></div>
+        <div class="w-80 mx-auto">
+            <div class="w-100 py-4">
+                <div class="w-100 pt-3 pb-5">
+                    <div v-for="answer in question.answers">
+                        <p><label class="radiocontainer">
+                            <span class="text-lg">{{answer.value}}</span>
+                            <input type="radio" name="answer" v-model="form.answer_id" :value="answer.id">
```
Comment (bumi) [inline app/javascript/views/Question.vue]: please note that we will have questions that will have more than one answer. - the user can select more than one answer... does that still work with a radiobutton.

Comment (musayann) [reply app/javascript/views/Question.vue]: Yeah you are right, I think we need to add an attribute like `multiple_choice`  in question, so in that case we wouldn't show radio button but instead checkboxes and we would also be able to override the validator which prevents to store an multiple answer belonging to the same question in `iteration_answer`

Comment (bumi) [reply app/javascript/views/Question.vue]: I had added a `multiple` and an `other` attribute on the question for such cases. 
so yeah if `question.multiple` is true we should allow the selection of multiple answers

File: app/models/iteration.rb [hunk 2]
```diff
@@ -19,7 +20,8 @@ def starting_question_id
 
   def completion_rate
     # just added a default before the implementing the function
-    12.0
+    Question.all.count
+    iteration_answers.count
```
Comment (bumi) [inline app/models/iteration.rb]: I think here is missing something... right now it is returning the `iteration_answers.count`

**Pull request merged**